proposal/package-importer.d.ts.md

```diff
@@ -0,0 +1,386 @@
+will support both the `"sass"` and the `"style"` conditions, as Sass can also
+use the CSS exports exposed through `"style"`. While in practice, `"style"`
+tends to be used solely for `css` files, we will support `scss`, `sass` and
+`css` files for either `"sass"` or `"style"`.
+
+## Types
+
+```ts
+import '../spec/js-api';
+```
+
+#### `usePkgImporter`
+
+If true, the compiler will use the built-in package importer to resolve any url
+with the `pkg` scheme. This importer follows Node logic to locate Sass files.
+
+Defaults to false.
+
+```ts
+declare module '../spec/js-api/options' {
+  interface Options<sync extends 'sync' | 'async'> {
+    /**
+     * Whether or not to enable the built-in package importer to resolve any url
+     * with the `pkg` scheme. This importer follows Node.js resolution logic.
+     *
+     * @defaultValue `false`
+     * @category Input
+     */
+    usePkgImporter?: boolean;
+  }
+}
+```
+
+## Semantics
+
+### Package Importers
+
+This proposal defines the requirements for Package Importers written by users or
+provided by implementations. It is a type of [Filesystem Importer] and will
+handle non-canonical URLs:
+
+- with the scheme `pkg`
+- followed by a package name
+- optionally followed by a path, with path segments separated with a forward
+  slash.
+
+The package name will often be the first path segment, but the importer should
+take into account any conventions in the environment. For instance, Node
+supports scoped package names, which start with `@` followed by 2 path segments.
+
+[filesystem importer]: ../spec/modules.md#filesystem-importer
+
+### Node Specific Semantics
+
+The Node package importer is an [importer] with an associated absolute `pkg:`
+URL named `base`. When the Node package importer is invoked with a string named
+`string` and a [previous url] `previousUrl`:
+
+- Let `url` be the result of [parsing `string` as a URL][parsing a url] with
+  `base` as the base URL. If this returns a failure, throw that failure.
+- If `url`'s scheme is not `pkg`, return null.
+- Let `resolved` be the result of [resolving a `pkg` URL] with `url` and `previousURL`
+- If `resolved` is null, return null.
+- Let `text` be the contents of the file at `resolved`.
+- Let `syntax` be:
+  - "scss" if `url` ends in `.scss`.
+  - "indented" if `url` ends in `.sass`.
+  - "css" if `url` ends in `.css`.
+  > The algorithm for [resolving a `file:` URL](../spec/modules.md#resolving-a-file-url)
+  > guarantees that `url` will have one of these extensions.
+- Return `text`, `syntax`, and `resolved`.
+
+[parsing a URL]: https://url.spec.whatwg.org/#concept-url-parser
+[resolving a `pkg` URL]: #resolving-a-pkg-url
+
+> Note that this algorithm does not automatically resolve index files, partials
+> or extensions, except where specified. When using the `pkg:` url scheme,
+> authors need to explicitly use the fully resolved filename.
+
+## Procedures
+
+### Resolving a `pkg` URL
```

Comment:
This algorithm name should make it clear that this is a Node.js-specific resolution algorithm, since we may have multiple built-in `pkg` algorithms in the future.